### DISCUSSION TITLE
Can I change connect method's header values in upstream mode?

### DISCUSSION THREAD

**Discussion by zhoulingmin** [mitmproxy/mitmproxy#5732] [Intercept Everything!]
When I use upstream mode, I want to add some headers in connect method. How can I do this? Thanks

**Comment by chribro88** [comment]
Should be able to change the upstream CONNECT headers with this: [https://docs.mitmproxy.org/stable/api/events.html#HTTPEvents.http_connect_upstream](https://docs.mitmproxy.org/stable/api/events.html#HTTPEvents.http_connect_upstream)

    def http_connect_upstream(self, flow: http.HTTPFlow) -> None:
        flow.request.headers['foo'] = "bar"
